ProjFS.Mac/PrjFSKext/Message_Kernel.cpp

```diff
@@ -33,7 +35,57 @@ void Message_Init(
     {
         header->pathSizeBytes = 0;
     }
-    
+
+    if (nullptr != fromPath)
+    {
+        header->fromPathSizeBytes = strlen(fromPath) + 1;
+    }
+    else
+    {
+        header->fromPathSizeBytes = 0;
+    }
+
     spec->messageHeader = header;
     spec->path = path;
+    spec->fromPath = fromPath;
+}
+
+uint32_t Message_EncodedSize(const Message& message)
+{
+    return sizeof(*message.messageHeader) + message.messageHeader->pathSizeBytes + message.messageHeader->fromPathSizeBytes;
+}
+
+uint32_t Message_Encode(void* buffer, const uint32_t bufferSize, const Message& message)
+{
+    uint8_t* bufferPosition = static_cast<uint8_t*>(buffer);
+    uint32_t bufferBytesRemain = bufferSize;
+    
+    assert(bufferSize >= sizeof(*message.messageHeader));
```

Comment:
We do a number of asserts for sizing here, which we didn't previously.  

I can see the value, in case an sizing error creeps in, but it does become more verbose.   @wilbaker

Reply:
Spoke with @wilbaker offline and we like your approach better than what was there previously.  It would be great if we can add tests for the new functions.